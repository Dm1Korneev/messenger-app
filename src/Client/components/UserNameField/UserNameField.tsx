import { TextField } from 'formik-material-ui';

type Props = Omit<
  React.ComponentPropsWithoutRef<typeof TextField>,
  'margin' | 'label' | 'fullWidth' | 'autoComplete'>

export const UserNameField = (props: Props) => (
  <TextField
    autoComplete="name"
    fullWidth
    label="Name *"
    margin="normal"
    {...props}
  />
);
