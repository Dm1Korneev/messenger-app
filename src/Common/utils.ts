
type Filter = Parameters<typeof Array.prototype.filter>[0]
export const distinct: Filter = (value, index, self) => self.indexOf(value) === index;
