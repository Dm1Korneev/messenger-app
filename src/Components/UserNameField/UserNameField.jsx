import React from 'react';
import { TextValidator } from 'react-material-ui-form-validator';
import PropTypes from 'prop-types';
import withStyles from '@material-ui/core/styles/withStyles';

function UserNameField(props) {
  const { value, onChange } = props;

  return (
    <TextValidator
      margin="normal"
      label="Name *"
      fullWidth
      onChange={onChange}
      name="name"
      id="name"
      autoComplete="name"
      color="primary"
      value={value}
      validators={['required']}
      errorMessages={['this field is required']}
    />
  );
}

UserNameField.defaultProps = {
  value: undefined,
};
UserNameField.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

const styles = () => ({});

export default withStyles(styles)(UserNameField);
