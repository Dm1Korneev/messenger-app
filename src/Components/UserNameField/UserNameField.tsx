import { TextField } from 'formik-material-ui';

type Props = Omit<
  React.ComponentPropsWithoutRef<typeof TextField>,
  'margin' | 'label' | 'fullWidth' | 'autoComplete'>

export const UserNameField = (props: Props) => (
  <TextField
    margin="normal"
    label="Name *"
    fullWidth
    autoComplete="name"
    {...props}
  />
);
