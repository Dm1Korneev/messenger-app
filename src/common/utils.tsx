export function distinct(value: any, index: number, self: Array<any>): boolean {
  return self.indexOf(value) === index;
}
