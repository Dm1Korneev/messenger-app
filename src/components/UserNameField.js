import React from "react";
import { withStyles } from "@material-ui/core/styles";
import { TextValidator } from "react-material-ui-form-validator";

function UserNameField(props) {
  const { value, onChange } = props;

  return (
    <TextValidator
      margin="normal"
      label="Name *"
      fullWidth
      onChange={onChange}
      name="name"
      id="name"
      autoComplete="name"
      color="primary"
      value={value}
      validators={["required"]}
      errorMessages={["this field is required"]}
    />
  );
}

const styles = theme => ({});

export default withStyles(styles)(UserNameField);
