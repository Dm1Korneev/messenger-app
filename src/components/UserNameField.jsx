import React from 'react';
import { TextValidator } from 'react-material-ui-form-validator';
import PropTypes from 'prop-types';

// material-ui
import withStyles from '@material-ui/core/styles/withStyles';

function UserNameField(props) {
  const { value, onChange } = props;

  return (
    <TextValidator
      margin="normal"
      label="Name *"
      fullWidth
      onChange={onChange}
      name="name"
      id="name"
      autoComplete="name"
      color="primary"
      value={value}
      validators={['required']}
      errorMessages={['this field is required']}
    />
  );
}

UserNameField.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

const styles = (theme) => ({});

export default withStyles(styles)(UserNameField);
