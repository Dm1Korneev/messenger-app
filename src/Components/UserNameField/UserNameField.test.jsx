import React from 'react';
import { render } from '@testing-library/react';

import { mockComponent } from 'testing/utils';

import UserNameField from './UserNameField';

jest.mock('react-material-ui-form-validator', () => ({
  TextValidator: mockComponent('TextValidator'),
}));

describe('render UserPasswordField component', () => {
  const props = {
    value: 'TEST',
    onChange: jest.fn(),
  };

  const setup = () => render(
    <UserNameField {...props} />,
  );

  test('snapshot test', async () => {
    const { asFragment } = setup();

    expect(asFragment()).toMatchSnapshot();
  });
});

