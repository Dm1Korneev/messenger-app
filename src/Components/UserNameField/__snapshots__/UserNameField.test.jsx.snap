// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render UserPasswordField component snapshot test 1`] = `
<DocumentFragment>
  <div
    data-originalcomponent="TextValidator"
    test-prop-autocomplete="\\"name\\""
    test-prop-color="\\"primary\\""
    test-prop-errormessages="[\\"this field is required\\"]"
    test-prop-fullwidth="true"
    test-prop-id="\\"name\\""
    test-prop-label="\\"Name *\\""
    test-prop-margin="\\"normal\\""
    test-prop-name="\\"name\\""
    test-prop-validators="[\\"required\\"]"
    test-prop-value="\\"TEST\\""
  />
</DocumentFragment>
`;
